instrumentation_sql.go

```diff
@@ -227,11 +227,38 @@ func (conn *wrappedSQLConn) ExecContext(ctx context.Context, query string, args
 }
 
 func (conn *wrappedSQLConn) CheckNamedValue(d *driver.NamedValue) error {
+	defer func() {
+		if err := recover(); err != nil {
+			sensor.logger.Debug("Swallowing error: ", err)
+		}
+	}()
+
 	if c, ok := conn.Conn.(driver.NamedValueChecker); ok {
 		return c.CheckNamedValue(d)
 	}
 
-	return nil
+	q := "create table instanatemp (a varchar(1))"
+
+	stmt, err := conn.Prepare(q)
```

Comment:
If driver.Conn doesn't implement `CheckNamedValue` we attempt to create a statement `driver.Stmt` and check if it implements the method `CheckNamedValue`.